src/color.cpp

```diff
@@ -170,7 +170,7 @@ static const named_color_t named_colors[] = {
 };
 
 wcstring_list_t rgb_color_t::named_color_names() {
-    size_t count = sizeof named_colors / sizeof *named_colors;
+    const size_t count = sizeof named_colors / sizeof *named_colors;
```

Comment:
So variables are marked const when they are initialized with `sizeof`? Or when they are effectively const and initialized with a constant expression?
There are some cases that suggest the former, but maybe your tool didn't catch those:

[`size_t max = 1024 * 1024 * 64;`](https://github.com/fish-shell/fish-shell/blob/master/src/lru.h#L300) (in a header)
[`size_t expected_evictee = 3;`](https://github.com/fish-shell/fish-shell/blob/master/src/fish_tests.cpp#L5529) and others in this test file, which is compiled separately from fish
[`size_t read_amt = 64 * 1024;`](https://github.com/fish-shell/fish-shell/blob/master/src/env_universal_common.cpp#L1295) which is compiled only on some systems

Reply:
The tool was treating this as a non constant expression for some reason.